tests/client_suite.go

```diff
@@ -1256,38 +1247,32 @@ func (s *ClientFunctionalSuite) Test_BufferedSignalCausesUnhandledCommandAndSche
 			}
 			receivedSig = sigVal
 		}
-
 		return nil
 	}
 
 	s.worker.RegisterWorkflow(workflowFn)
 
-	id := "functional-test-unhandled-command-new-task"
 	workflowOptions := sdkclient.StartWorkflowOptions{
-		ID:        id,
-		TaskQueue: s.taskQueue,
+		ID:        tv.WorkflowID(),
+		TaskQueue: tv.TaskQueue().Name,
 		// Intentionally use same timeout for WorkflowTaskTimeout and WorkflowRunTimeout so if workflow task is not
 		// correctly dispatched, it would time out which would fail the workflow and cause test to fail.
 		WorkflowTaskTimeout: 10 * time.Second,
 		WorkflowRunTimeout:  10 * time.Second,
 	}
-	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-	defer cancel()
 	workflowRun, err := s.sdkClient.ExecuteWorkflow(ctx, workflowOptions, workflowFn)
 	if err != nil {
 		s.Logger.Fatal("Start workflow failed with err", tag.Error(err))
 	}
 
 	s.NotNil(workflowRun)
 	s.True(workflowRun.GetRunID() != "")
+	tv = tv.WithRunID(workflowRun.GetRunID())
 
 	// block until first workflow task started
-	select {
-	case <-sigReadyToSendChan:
-	case <-ctx.Done():
```

Comment:
I _think_ by omitting `<-ctx.Done` (here and above) the test _might_ get stuck and run until the GHA terminates.

Reply:
Yep! This is a recipe for a deadlock

Reply:
Yep, reinstated, thanks. (The other `select` that I removed I believe is not necessary).